lua/indentmini/init.lua

```diff
@@ -46,11 +55,14 @@ local function indentline()
       indent = prev > 20 and 4 or prev
     end
 
-    local hi_name = hl_group()
-
     ctx[row] = indent
 
-    for i = 1, indent - 1, indent_step(bufnr) do
+    local shiftw = indent_step(bufnr)
+    for i = 1, indent - 1, shiftw do
```

Comment:
we can do something like

```lua
    for loop do
         local level = fn.indent(row + 1) -- row is 0-indexed
         local hi = ('IndentLine%d'):format(level)
        api.nvim_set_hl(0, hi , { link = 'IndentLine', default = true})
```

Reply:
Okay, I was able to use that, and the extra logic here is to make it so that the colors will repeat once the indent level is more than how many colors have been defined.

This tries to check if there's highlight groups defined for a given level, if not then it will re-use a previously defined one, and if none are defined, it uses the default `IndentLine` group. 

Let me know what you think.